packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -828,6 +832,31 @@ export class Repository extends RepositoryBase {
   }
 }
 
+function isValidEcrRepoArn(arn: string): boolean {
+  const arnRegex = /^arn:(\w+):ecr:([a-z]+-[a-z]+-\d{1}):(\d{12}):repository\/([A-Za-z]+.*)$/;
+  const arnInfo = arnRegex.exec(arn);
+
+  if (arnInfo === null) {
+    throw new Error(`There was an error parsing repository arn: ${arn}.`);
```

Comment:
It would return `null` if the regex match fails which would mean that the arn does not align with the regex. 

More here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#return_value


I can change the error to say,
```
The repository arn does not match the regex and is invalid: ${repositoryArn}.
```